godotproject/enemy-Ant.gd

```diff
@@ -82,7 +82,7 @@ func _on_PlayerDetector_body_entered(body):
 func _on_ActionDelay_timeout():
 	if attackState == STATE.detected_player_e:
 		launch_attack()
-	elif attackState == STATE.attacking_player_e:
+	elif attackState == STATE.attacking_player_e || attackState == STATE.idle_e:
 		attackState = STATE.idle_e
```

Comment:
I dont really understand the reason for adding this OR. If the attackState is idle, why reset it?

Reply:
There's more code inside the body of the elif after this, responsible for trying to attack the player. I think it'll make it so that the player is always detected no matter what, when the ActionDelay timer goes off